### PR TITLE
Tmedia 254 header nav chain

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Focus on header nav chain for primary-font refactor 

## Jira Ticket
- [TMEDIA-254](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-254&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria
- Uses primaryfont shared styles not getThemeStyle
- Existing behavior is unchanged

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-254-header-nav-chain`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/header-nav-chain`
3. Go to http://localhost/pf/allblockspr/?_website=the-gazette
4. Ensure text and markers are white 

Screen Shot 2021-07-15 at 09 29 47

### Before
- Creating redundant styled components for primary font 

### After
- Less duplication of styles

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by JackHowa** [comment]
Updated with fix for sections button primary font 

before: 

Screen Shot 2021-07-15 at 09 36 47
Screen Shot 2021-07-15 at 09 35 53


after: 

Screen Shot 2021-07-15 at 09 39 11


______

Found a couple more primary font usages in header nav chain

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx [hunk 1]
```diff
@@ -70,15 +69,9 @@ const StyledNav = styled.nav`
...
-    font-family: ${(props) => props.font};
-  }
-
 `;
 
 const StyledSectionDrawer = styled.div`
-  font-family: ${(props) => props.font};
   z-index: ${sectionZIdx};
 
   @media screen and (max-width: ${(props) => props.breakpoint}px) {
```
[the remark this answers is not in the document]
Comment (JackHowa) [reply blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx:77]: before: 

129 problems found
 severity level "error": 125
  no-descending-specificity: 25
  selector-max-compound-selectors: 45
  scss/selector-no-redundant-nesting-selector: 10
  max-nesting-depth: 2
  rule-empty-line-before: 3
  selector-no-qualifying-type: 14
  selector-class-pattern: 6
  color-hex-case: 1
  declaration-property-value-disallowed-list: 2
  block-opening-brace-space-before: 1
  selector-max-id: 9
  **media-feature-name-no-unknown: 6**
  selector-list-comma-newline-after: 1
 severity level "warning": 4
  plugin/no-unsupported-browser-features: 4



after: 

123 problems found
 severity level "error": 119
  no-descending-specificity: 25
  selector-max-compound-selectors: 45
  scss/selector-no-redundant-nesting-selector: 10
  max-nesting-depth: 2
  rule-empty-line-before: 3
  selector-no-qualifying-type: 14
  selector-class-pattern: 6
  color-hex-case: 1
  declaration-property-value-disallowed-list: 2
  block-opening-brace-space-before: 1
  selector-max-id: 9
  selector-list-comma-newline-after: 1
 severity level "warning": 4
  plugin/no-unsupported-browser-features: 4

[the remark this answers is not in the document]
Comment (nschubach) [reply blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx:77]: I assume it's complaining about the `${(props) => props.breakpoint}` ? the rest looks fine.

**Review by nschubach — APPROVED** [review]

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (JackHowa) [reply blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx:77]: ya I fixed it in a separate stylelint pr but good shoutout https://github.com/WPMedia/fusion-news-theme-blocks/pull/980

**Pull request merged**